### PR TITLE
Adding GetKnownUsers api endpoint documentation

### PR DESCRIPTION
**Pull request by jespino**

### CODE REVIEW

File: v4/source/users.yaml [hunk 1]
```diff
@@ -754,6 +754,40 @@
...
+        - lang: Go
+          source: |
+            import "github.com/mattermost/mattermost-server/model"
+
+            Client := model.NewAPIv4Client("https://your-mattermost-url.com")
+            Client.Login("email@domain.com", "Password1")
+
+            userID := "4xp9fdt77pncbef59f4k1qe83o"
+
+            userIds, resp := Client.GetKnownUsers(userID)
```
Comment (streamer45) [inline v4/source/users.yaml:789]: `Client.GetKnownUsers()` does not take any argument. The user is taken from the session if I recall correctly.

Comment (jespino) [reply v4/source/users.yaml:789]: Yes, copy and paste :facepalm:

File: v4/source/users.yaml [hunk 2]
```diff
@@ -754,6 +754,38 @@
...
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/UsersStats"
+        "401":
+          $ref: "#/components/responses/Unauthorized"
+      x-code-samples:
+        - lang: Go
+          source: |
+            import "github.com/mattermost/mattermost-server/model"
```
Comment (streamer45) [inline v4/source/users.yaml:782]: Actually created https://mattermost.atlassian.net/browse/MM-24438 to address this. `v5` should be used in the import path.

Comment (jespino) [reply v4/source/users.yaml:782]: Very good point

File: v4/source/users.yaml [hunk 3]
```diff
@@ -754,6 +754,38 @@
             if ($resp->getStatusCode() == 200) {
                 $users = json_decode($resp->getBody());
             }
+  /users/known:
+    get:
+      tags:
+        - users
+      summary: Get user ids of known users
```
Comment (justinegeffen) [inline v4/source/users.yaml:761]:
```suggestion
      summary: Get user IDs of known users
```

File: v4/source/users.yaml [hunk 4]
```diff
@@ -754,6 +754,38 @@
...
+      description: |
+        Get the list of user ids of users with any direct relationship with a
```
Comment (justinegeffen) [inline v4/source/users.yaml:763]:
```suggestion
        Get the list of user IDs of users with any direct relationship with a
```

File: v4/source/users.yaml [hunk 5]
```diff
@@ -754,6 +754,38 @@
...
+        user. That means any user sharing any channel, including direct and
+        group channels.
+        ##### Permissions
+        Must be authenticated.
+
+        __Minimum server version__: 5.23
+      responses:
+        "200":
+          description: Known users ids retrieval successful
```
Comment (justinegeffen) [inline v4/source/users.yaml:772]:
```suggestion
          description: Known users' IDs retrieval successful
```

**Pull request merged**